### PR TITLE
Design feedback & sanitize / validate text records BAPP-262

### PR DESCRIPTION
**Pull request by kirkas**
https://linear.app/coinbase/issue/BAPP-262/customize-profile-add-validation-on-social-text-records-inputs

Validate textRecords

image

Sanitize textRecords

image

Tidy up modal:
image

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/hooks/useReadBaseEnsTextRecords.ts [hunk 1]
```diff
@@ -51,11 +51,12 @@ export default function useReadBaseEnsTextRecords({
...
 
   const getExistingTextRecords = useCallback(async () => {
+    console.log('GET', { address, username });
     if (!username) return;
     const result = await getBasenameTextRecords(username);
     return result;
-  }, [username]);
-
+  }, [address, username]);
+  console.log('GET', { address, username });
```
Comment (zencephalon) [inline apps/web/src/hooks/useReadBaseEnsTextRecords.ts:59]: Ditto

File: apps/web/src/utils/usernames.ts [hunk 2]
```diff
@@ -84,10 +84,24 @@ export const textRecordsSocialFieldsEnabledIcons: Partial<Record<UsernameTextRec
...
+    // Handle isn't a url, no problem
   }
-  return handle.replace(/^@/, '');
+
+  // remove the '@' if present
+  if (handleSanitized.startsWith('@')) {
+    handleSanitized = handleSanitized.substring(1);
+  }
+
+  return handleSanitized.replace(/^@/, '');
```
Comment (JFrankfurt) [inline apps/web/src/utils/usernames.ts:104]: this reads like duplicate logic to me. Am I dumb?

File: apps/web/src/hooks/useReadBaseEnsTextRecords.ts (hunk 1 above)
Comment (JFrankfurt) [reply apps/web/src/hooks/useReadBaseEnsTextRecords.ts:59]:
```suggestion
```

File: apps/web/src/components/Basenames/UsernameProfileSidebar/index.tsx [hunk 3]
```diff
@@ -38,7 +38,7 @@ export default function UsernameProfileSidebar() {
   });
 
   const textRecordKeywords = existingTextRecords[UsernameTextRecordKeys.Keywords];
-
+  console.log({ existingTextRecords, textRecordKeywords });
```
[the remark this answers is not in the document]
Comment (JFrankfurt) [reply apps/web/src/components/Basenames/UsernameProfileSidebar/index.tsx:41]:
```suggestion
```

File: apps/web/src/utils/usernames.ts (hunk 2 above)
Comment (kirkas) [reply apps/web/src/utils/usernames.ts:104]: no i am  :)

**Review by zencephalon — APPROVED** [review]

**Pull request merged**